converter_app/converters.py

```diff
@@ -58,22 +60,97 @@ def _prepare_identifier(self):
         profile_identifiers = self.profile.data.get('identifiers', [])
         self.identifiers = []
         for identifier in profile_identifiers:
-            if identifier.get('outputTableIndex') is None:
+            output_table_index = identifier.get('outputTableIndex')
+            if output_table_index is None:
                 # if no outputTableIndex was set this identifier is valid for every table
                 # no adjustment has to be done
                 self.identifiers.append(identifier)
             else:
-                # adjust this identifier for every input table
-                for input_table_index, _ in enumerate(self.input_tables):
-                    # make a copy of the identifier and adjust the outputTableIndex
-                    identifier_copy = copy.deepcopy(identifier)
-                    identifier_copy['outputTableIndex'] = input_table_index
+                if self._has_loop(output_table_index):
+                    # adjust this identifier for every input table
+                    for input_table_index, _ in enumerate(self.input_tables):
+                        if not self._check_loop_condition(output_table_index, input_table_index):
+                            continue
+                        # make a copy of the identifier and adjust the outputTableIndex
+                        identifier_copy = copy.deepcopy(identifier)
+                        identifier_copy['outputTableIndex'] = (input_table_index
+                                                               + self._get_output_table_index(identifier['outputTableIndex']))
+
+                        # adjust the (input)tableIndex as well if it was not null
+                        if identifier_copy.get('tableIndex') is not None:
+                            identifier_copy['tableIndex'] = input_table_index
+
+                        self.identifiers.append(identifier_copy)
+                else:
+                    identifier['outputTableIndex'] = self._get_output_table_index(identifier['outputTableIndex'])
+                    self.identifiers.append(identifier)
 
-                    # adjust the (input)tableIndex as well if it was not null
-                    if identifier_copy.get('tableIndex') is not None:
-                        identifier_copy['tableIndex'] = input_table_index
+    def _has_loop(self, index):
+        if len(self.profile_output_tables) <= index:
+            return False
+        if self.profile_output_tables[index].get('loopType') == 'all':
+                return self.profile_output_tables[index].get('matchTables')
+        else:
+            loop_header = self.profile_output_tables[index].get('table').get('loop_header')
```

Comment:
Reduce usage of .get() if table is not in elf.profile_output_tables[index] it breaks anyhow

table = self.profile_output_tables[index]['table']